tests/components/here_travel_time/test_config_flow.py

```diff
@@ -0,0 +1,550 @@
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["errors"] == {}
+
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {
+            CONF_API_KEY: API_KEY,
+            CONF_MODE: TRAVEL_MODE_CAR,
+            CONF_NAME: "test",
+        },
+    )
+    await hass.async_block_till_done()
+
+    assert result2["type"] == data_entry_flow.RESULT_TYPE_MENU
+    assert result2["menu_options"] == menu_options
+
+
+@pytest.mark.usefixtures("valid_response")
+async def test_step_origin_coordinates(
+    hass: HomeAssistant, user_step_result: data_entry_flow.FlowResult
+) -> None:
+    """Test the origin coordinates step."""
+    menu_result = await hass.config_entries.flow.async_configure(
+        user_step_result["flow_id"], {"next_step_id": "origin_coordinates"}
+    )
+    assert menu_result["type"] == data_entry_flow.RESULT_TYPE_FORM
+
+    location_selector_result = await hass.config_entries.flow.async_configure(
+        menu_result["flow_id"],
+        {
+            "origin": {
+                "latitude": float(CAR_ORIGIN_LATITUDE),
+                "longitude": float(CAR_ORIGIN_LONGITUDE),
+                "radius": 3.0,
+            }
+        },
+    )
+    assert location_selector_result["type"] == data_entry_flow.RESULT_TYPE_MENU
+
+
+@pytest.mark.usefixtures("valid_response")
+async def test_step_origin_entity(
+    hass: HomeAssistant, user_step_result: data_entry_flow.FlowResult
+) -> None:
+    """Test the origin coordinates step."""
+    menu_result = await hass.config_entries.flow.async_configure(
+        user_step_result["flow_id"], {"next_step_id": "origin_entity"}
+    )
+    assert menu_result["type"] == data_entry_flow.RESULT_TYPE_FORM
+
+    entity_selector_result = await hass.config_entries.flow.async_configure(
+        menu_result["flow_id"],
+        {"origin_entity_id": "zone.home"},
+    )
+    assert entity_selector_result["type"] == data_entry_flow.RESULT_TYPE_MENU
+
+
+@pytest.mark.usefixtures("valid_response")
+async def test_step_destination_coordinates(
+    hass: HomeAssistant, origin_step_result: data_entry_flow.FlowResult
+) -> None:
+    """Test the origin coordinates step."""
+    menu_result = await hass.config_entries.flow.async_configure(
+        origin_step_result["flow_id"], {"next_step_id": "destination_coordinates"}
+    )
+    assert menu_result["type"] == data_entry_flow.RESULT_TYPE_FORM
+
+    location_selector_result = await hass.config_entries.flow.async_configure(
+        menu_result["flow_id"],
+        {
+            "destination": {
+                "latitude": float(CAR_ORIGIN_LATITUDE),
+                "longitude": float(CAR_ORIGIN_LONGITUDE),
+                "radius": 3.0,
+            }
+        },
+    )
+    assert location_selector_result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
```

Comment:
I think it also important to test the resulting config entry as well here, to make sure the final result looks good and nothing was dropped at any of the individual steps.  Want to add some coverage on the two destination steps?

I realize there are now many different combinations that are possible, but i think just having some basic coverage end to end should be fine.

Reply:
Thanks for the catch. Meant to do it and forgot about it!